<font size="10">Video Game Sales Case Study</font>

<font size="5">Overview:</font>

Data Sourced: https://www.kaggle.com/datasets/gregorut/videogamesales

Today, I will discuss/analyze a dataset about the list of video games that have more than 100,000 copies sold. I picked this dataset because video games are one of my favorite hobbies, and I know that other people enjoy video games as a pasttime as well. Also, this dataset shows the ranks of these video games that have more than 100,000 copies sold, what console the game is for, the year it was released, genre, sales(in millions), etc. It provides a lot of interesting information that I can analyze and try to make a correlation between copies sold and some other variable. Besides this, the analysis will be helpful to fans of video games and gaming copmanies to see the top video game sales, and what type of factors make a game successful in terms of sales.

In [261]:
import pandas as pd
import numpy as np
import plotly
import plotly.express as px
import plotly.graph_objects as go

# plotly.io is a low-level interface for interacting with figures/
# plotly.io.templates lists available plotly templates
# https://plotly.com/python-api-reference/plotly.io.html
import plotly.io as pio

pd.set_option('display.max_columns', 20)

In [262]:
import pandas as pd

url = 'https://raw.githubusercontent.com/themuid/Final-Case-Study-Project/261a95bd2679309790bb65751c4e1f2e466da792/vgsales.csv'
df = pd.read_csv(url)

# Dataset is now stored in a Pandas Dataframe

<font size="5">Questions to answer today:</font>



1.   What can we say about the top 10 video games sold in terms of JP Sales?
2.   How many top video games in terms of sales were released in the year 2003?
3.   What can we say about the distribution of global_sales for these video games?
4.   What is the year in which the least amount of top selling video games released in a year since 2000, and what is that number?
5.   What can we infer about a heatmap comparing the "Year" and "Genre" variables for Video Game Sales?
6.   What can a heatmap made with our dataset tell us about Video Game Sales?







<font size="5">Exploratory Data Analysis (EDA) </font>

In [263]:
df = pd.read_csv('https://raw.githubusercontent.com/themuid/Final-Case-Study-Project/261a95bd2679309790bb65751c4e1f2e466da792/vgsales.csv')
df

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

Before we proceed further, here are some important things to note from the data sourced, in particular, the meaning of our columns:

* Rank - Ranking of overall sales

* Name - The game name

* Platform - Platform of the games release (i.e. PC, PS4, XB, Wii, DS, etc.)

* Year - Year of the game's release

* Genre - Genre of the game

* Publisher - Publisher of the game

* NA_Sales - Sales in North America (in millions)

* EU_Sales - Sales in Europe (in millions)

* JP_Sales - Sales in Japan (in millions)

* Other_Sales - Sales in the rest of the world (in millions)

* Global_Sales - Total worldwide sales.

Now that we have our data, we can display it in various ways, using df.head(), df.tail() and df.sample():

In [264]:
df.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [265]:
df.tail()

Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  
16595          0.0          0.01  
16596          0.0          0.01  
16597          0.0          0.01

In [266]:
df.sample(1)

Rank                                      Name Platform    Year  \
10991  10993  Jikkyou Powerful Pro Yakyuu 14 Ketteiban      PS2  2007.0   

        Genre                     Publisher  NA_Sales  EU_Sales  JP_Sales  \
10991  Sports  Konami Digital Entertainment       0.0       0.0      0.09   

       Other_Sales  Global_Sales  
10991          0.0          0.09

<font size="5">Note:</font>

The df.head() function will show the top rows of our data, whereas the df.tail() function shows the bottommost rows of our data. 

The df.sample() function will show a random part of our date, changing each time when we run this function. 

The number we put in these parantheses will be how much rows it will display. 
As a default, if no value inserted in the parantheses, there will be 5 rows dislayed for df.head() and df_tail, and 1 row displayed for df.sample()

For our data set, every column is a relevant variable to analyzing our video game sales ranking, so no need to run any code to remove/drop them. The author who orignally posted the data also removed any incomplete information, which saves us some time and code. We can start by using the df.info() function to check the data of each column, and see the number of rows with non-missing values.

In [267]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


Now, we can also confirm the number of columns and rows in the Video Games Sales data by using the df_shape and print functions.

In [268]:
num_rows, num_cols = df.shape
print(f'There are {num_rows} rows and {num_cols} columns in the dataset.')

There are 16598 rows and 11 columns in the dataset.


<font size="5">1st Question:</font>

Now, we will aim to use (EDA) to answer our 1st question of the Case Study.

"What can we say about the top 10 video games sold in terms of JP Sales?"

In [269]:
df_videogame_sales = df[['Rank', 'Name', 'Platform', 'Year', 'Genre', 'Publisher', 'NA_Sales', 'EU_Sales', 'JP_Sales', 'Other_Sales', 'Global_Sales']].head(20)
df_videogame_sales.sort_values('JP_Sales', ascending= False, inplace=True)
df_top_videogame_sales = df_videogame_sales.rename(columns={'Global_Sales':'Worldwide_Sales'})

df_top_videogame_sales

Rank                                          Name Platform    Year  \
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
1      2                             Super Mario Bros.      NES  1985.0   
6      7                         New Super Mario Bros.       DS  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
5      6                                        Tetris       GB  1989.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
2      3                                Mario Kart Wii      Wii  2008.0   
0      1                                    Wii Sports      Wii  2006.0   
13    14                                       Wii Fit      Wii  2007.0   
18    19                             Super Mario World     SNES  1990.0   
3      4                             Wii Sports Resort      Wii  2009.0   
7      8                                      Wii Play      Wii  2006.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
10    11                                    Nintendogs       DS  2005.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
9     10                                     Duck Hunt      NES  1984.0   
15    16                            Kinect Adventures!     X360  2010.0   

           Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
4   Role-Playing                Nintendo     11.27      8.89     10.22   
12  Role-Playing                Nintendo      9.00      6.18      7.20   
1       Platform                Nintendo     29.08      3.58      6.81   
6       Platform                Nintendo     11.38      9.23      6.50   
8       Platform                Nintendo     14.59      7.06      4.70   
5         Puzzle                Nintendo     23.20      2.26      4.22   
19          Misc                Nintendo      4.75      9.26      4.16   
11        Racing                Nintendo      9.81      7.57      4.13   
2         Racing                Nintendo     15.85     12.88      3.79   
0         Sports                Nintendo     41.49     29.02      3.77   
13        Sports                Nintendo      8.94      8.03      3.60   
18      Platform                Nintendo     12.78      3.75      3.54   
3         Sports                Nintendo     15.75     11.01      3.28   
7           Misc                Nintendo     14.03      9.20      2.93   
14        Sports                Nintendo      9.09      8.59      2.53   
10    Simulation                Nintendo      9.07     11.00      1.93   
16        Action    Take-Two Interactive      7.01      9.27      0.97   
17        Action    Take-Two Interactive      9.43      0.40      0.41   
9        Shooter                Nintendo     26.93      0.63      0.28   
15          Misc  Microsoft Game Studios     14.97      4.94      0.24   

    Other_Sales  Worldwide_Sales  
4          1.00            31.37  
12         0.71            23.10  
1          0.77            40.24  
6          2.90            30.01  
8          2.26            28.62  
5          0.58            30.26  
19         2.05            20.22  
11         1.92            23.42  
2          3.31            35.82  
0          8.46            82.74  
13         2.15            22.72  
18         0.55            20.61  
3          2.96            33.00  
7          2.85            29.02  
14         1.79            22.00  
10         2.75            24.76  
16         4.14            21.40  
17        10.57            20.81  
9          0.47            28.31  
15         1.67            21.82

In [287]:
fig = px.bar(
    df_top_videogame_sales, 
    x='Name', 
    y='JP_Sales',
    color = 'Name',
    title='Top 20 Video Game Sales In Japan',
    text = 'JP_Sales',
    width = 900,
    height = 900
)
fig.show()

<font size="5">Takeaways:</font>

By analyzing our graph, we can see that the top video game sold in terms of Japan Sales are Pokemon games, making sense due to the fandom/popularity/reach it has over there. We can also see that near the bottom, there are more violent games such as Grand Theft Auto that might not be bought/played as much in Japan.

<font size="5">2nd Question:</font>

Now, we will aim to use (EDA) to answer our 2nd question of the Case Study.

"How many top video games in terms of sales were released in the year 2003?"

<font size="5">Key Reminder:</font>

Before we do our initial code, in order to compare "Year" and "Name" variables, they both have to be the same data type. From the df.info() ran earlier in this projectm we know that "Year" is "float64", while "Name is an "object". Thus, we just have to run some code to make it so that they're both objects.

In [271]:
df['Year'] = df['Year'].astype(str)

Now, we can run code to compare the two, since they're the same data type.

In [272]:
df_videogame_sales_by_year = df.groupby('Year', as_index = False).agg({
    'Name': 'count',
})

df_videogame_sales_by_year.rename(columns={'Name':'Num_VideoGames'}, inplace = True)
df_videogame_sales_by_year.sort_values(['Num_VideoGames'], ascending = False, inplace= True)

df_videogame_sales_in_2000s = df_videogame_sales_by_year[(df_videogame_sales_by_year['Year']>= '2000.0')]

df_videogame_sales_in_2000s

Year  Num_VideoGames
29  2009.0            1431
28  2008.0            1428
30  2010.0            1259
27  2007.0            1202
31  2011.0            1139
26  2006.0            1008
25  2005.0             941
22  2002.0             829
23  2003.0             775
24  2004.0             763
32  2012.0             657
35  2015.0             614
34  2014.0             582
33  2013.0             546
21  2001.0             482
20  2000.0             349
36  2016.0             344
39     nan             271
37  2017.0               3
38  2020.0               1

In [273]:
fig = px.treemap(
    df_videogame_sales_in_2000s,
    path= ['Year'],
    title='Number of Top Selling Video Games Released in 2000s Breakdown',
    values='Num_VideoGames',
    height=1000
)

fig.show()

<font size="5">Takeaways:</font>

By analyzing our visualization, we can see that in the year 2003, there were 775 games that all sold over 100,000 copies. We also see the amount of top selling video games sold in other years of the 2000s.

<font size="5">3rd Question:</font>

Now, we will aim to use (EDA) to answer our 3rd question of the Case Study.

"What can we say about the distribution of global_sales for these video games?"

In [274]:
fig = px.box(
    df, 
    x='Global_Sales',
    orientation='h',
    title='Video Game Global Sales Distribution (Horizontal)',
    width=16000,
    template = 'ggplot2'
)
fig.show()

<font size="5">Takeaways:</font>

By analyzing our visualization, we can see that in our data set, for Global_Sales, the min copies sold for any video game is 0.01, the max copies sold for any video game is 82.74, and the median copies sold for any video game is 0.17. As a reminder, these numbers are all in terms of millions. 

<font size="5">4th Question:</font>

Now, we will aim to use (EDA) to answer our 4th question of the Case Study.

"What is the year in which the least amount of top selling video games released in a year since 2000, and what is that number?"

In [275]:
fig = px.histogram(
    df, 
    x='Year',
    title = 'Number of Top Selling Video Games By Year',
    nbins = 80,
    height = 3000,
    template='plotly_dark'
)
fig.show()

<font size="5">Takeaways:</font>

By analyzing our visualization, we can see that since the year 2000, the year with the least amount of video games selling over 100,000 copies is the year 2020, with only 1 game. This makes sense, as since the game has been released so recently, it hasn't had enough time to sell as many copies as games released earier in this dataset.

<font size="5">Extra Tips:</font>

We can also run some more code to find this the name of this 1 game in 2020.

In [276]:
df = pd.read_csv('https://raw.githubusercontent.com/themuid/Final-Case-Study-Project/261a95bd2679309790bb65751c4e1f2e466da792/vgsales.csv')

year_2020 = df[df['Year'] == 2020]

year_2020


Rank                    Name Platform    Year       Genre Publisher  \
5957  5959  Imagine: Makeup Artist       DS  2020.0  Simulation   Ubisoft   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
5957      0.27       0.0       0.0         0.02          0.29

From our code, we know this 1 game from 2020 that sold over 100,000 copies worldwide is "Imagine: Makeup Artist"

<font size="5">5th Question:</font>

Now, we will aim to use (EDA) to answer our 5th question of the Case Study.

"What can we infer about a heatmap comparing the "Year" and "Genre" variables for Video Game Sales?"

In [277]:
fig = px.density_heatmap(
    df, 
    x='Year', 
    y='Genre',
    title='Video Game Global Sales Distribution (Horizontal)'
    )
fig.show()

<font size="5">Takeaways:</font>

By analyzing our visualization, we can see that in the period between 2008-2009, the genre which had the most video games selling over 100,000 copies is "Action", with 493 games. We can also see that the most successful genres for video game sales are Adventure, Action, Shooter, and Sports, as they glow brightest on this heatmap. meaning the count is higher.

<font size="5">6th Question:</font>

Now, we will aim to use (EDA) to answer our 6 question of the Case Study.

"What can a heatmap made with our dataset tell us about Video Game Sales?"

In [278]:
fig = px.sunburst(
    df,
    path = ['Platform', 'Genre'],
    values = 'Global_Sales',
    height = 1000,
    title = 'Video Game Sales by Platform and Genre'
)
fig.show()


<font size="5">Takeaways:</font>

By analyzing our visualization, we can see that the Platform/Console with the most video game sales is the PS2 with 1,255.64 in Global Sales. Remember that these numbers are expressed in millions of copies. This analysis makes sense as the PS2 is one of the oldest consoles out, with many fun games to play on, and creating a lot of nostalgia for the older generation. It makes sense that it has the most video game sales.

<font size="5">Conclusion/Summary:</font>

**All these points can be inferred from the Exploratory Data Analysis (EDA) performed in this project.**

*   Pokemon Red/Pokemon Blue, and Pokemon Gold/Silver are included in the top 20 games sales with over 100,000 copies in Japan.
*   There were 775 games released in 2003 that all ended up selling over 100,000 copies. The year with the most games released selling over 100,000 copies is 2009, with 1431 games. 
*   For Global_sales, the minimum copies sold for any video game is 0.01, the maximum copies sold for any video game is 82.74, and the median copies sold for any video game is 0.17. All of these are expressed in millions.
*   The year 2020 has the least amount of video games selling over 100,000 copies released, with only 1 game, named "Imagine: Makeup Artist".
*   In the period between 2008-2009, the genre which had the most video games selling over 100,000 copies is "Action", with 493 games. Besides this, the most successful genres for video game sales are Adventure, Action, Shooter, and Sports, as they glow brightest on this heatmap. meaning the count is higher.
*   The Platform/Console with the most video game sales is the PS2 with 1,255.64 in Global Sales. All of these are expressed in millions. Some other Platforms with high video game sales besides the PS2 are the X360, PS3, Wii, and PS. 

In [285]:
!jupyter nbconvert --to html Mui,Darwin_FinalProject.ipynb

[NbConvertApp] Converting notebook Mui,Darwin_FinalProject.ipynb to html
[NbConvertApp] Writing 1219296 bytes to Mui,Darwin_FinalProject.html
